frontend/src/app/history/download-history/download-history.component.html

```diff
@@ -48,6 +55,16 @@ <h1 class="title">Downloads</h1>
                         <span>download file</span>
                     </a>
                 </td>
+                <td>
+                    <a *ngIf="download.status === 'done' || download.status ==='error'" role="button" tabindex="0"
+                        (click)="deleteDownload(download)"
+                        (keydown.enter)="deleteDownload(download)">
+                        <span class="icon">
+                            <fa-icon [icon]="faTrash"></fa-icon>
+                        </span>
+                        <span>delete download</span>
+                    </a>
+                </td>
```

Comment:
Nice.

Addition my earlier comment: if you choose to make the search history menu more like this page, don't bother adding this action. The search history includes a unique filter, which means deleting one query would usually cause another identical one to appear. A decent delete button per query would take more time.